dhcpv6/modifiers_test.go

```diff
@@ -46,32 +46,30 @@ func TestWithRequestedOptions(t *testing.T) {
 	oro := opt.(*OptRequestedOption)
 	require.ElementsMatch(t, oro.RequestedOptions(), []OptionCode{OptionClientID})
 	// Check if already set options are preserved
-	WithRequestedOptions(OptionServerID)(m)
+	m = WithRequestedOptions(OptionServerID)(m)
 	opt = m.GetOneOption(OptionORO)
 	require.NotNil(t, opt)
 	oro = opt.(*OptRequestedOption)
 	require.ElementsMatch(t, oro.RequestedOptions(), []OptionCode{OptionClientID, OptionServerID})
 }
 
 func TestWithIANA(t *testing.T) {
-	var d Message
-	WithIANA(OptIAAddress{
+	d := WithIANA(OptIAAddress{
 		IPv6Addr:          net.ParseIP("::1"),
 		PreferredLifetime: 3600,
 		ValidLifetime:     5200,
-	})(&d)
-	require.Equal(t, 1, len(d.Options))
-	require.Equal(t, OptionIANA, d.Options[0].Code())
+	})(&Message{})
+	require.Equal(t, 1, len(d.(*Message).Options))
```

Comment:
Now you have to do all these casts, and that's crummy. I'd prefer to separate Modifiers and Encaps rather than reintroduce the need for casting.

Reply:
In my mind, casting like this will only lead to production panics and reboots.

Reply:
> Now you have to do all these casts

I am reverting as few things as possible to get some internal code to work again.
These casts were previously there.

> In my mind, casting like this will only lead to production panics and reboots.

I agree but these are only tests.

Reply:
> > Now you have to do all these casts
> 
> I am reverting as few things as possible to get some internal code to work again.
> These casts were previously there.
> 
> > In my mind, casting like this will only lead to production panics and reboots.
> 
> I agree but these are only tests.

There is no such thing as "only tests". These tests use exported features of the library. That means any user that wants to use the library has to cast like this to make it work. Yes, it was previously there, but that's the reason for the improvement.